libraries/ashrae/guideline36/4.11-hot-water-plant.ttl

```diff
@@ -0,0 +1,434 @@
+    sh:or (
+        sh:node :constant-primary-constant-secondary-loop
+        sh:node :constant-primary-variable-secondary-loop
+        sh:node :variable-primary-constant-secondary-loop
+        sh:node :variable-primary-variable-secondary-loop
+    ) ;
+    # applications, optional per 4.11.3
+    sh:property [
+        rdfs:label "HW supply temperature" ;
+        rdfs:comment "Optional for primary-secondary plants." ;
+        sh:path brick:hasPoint ;
+        sh:qualifiedValueShape [ sh:class brick:Hot_Water_Supply_Temperature_Sensor ] ;
+        sh:qualifiedMinCount 0 ;
+        sh:qualifiedMaxCount 1 ;
+    ] ;
+    # required
+    sh:property [
+        rdfs:label "Secondary HW supply temperature" ;
+        sh:path brick:hasPoint ;
+        sh:qualifiedValueShape [ sh:class brick:Hot_Water_Supply_Temperature_Sensor ] ;
+        sh:qualifiedMinCount 1 ;
+        sh:qualifiedMaxCount 1 ;
+    ] ;
+    # required
+    sh:property [
+        rdfs:label "Secondary HW return temperature" ;
+        sh:path brick:hasPoint ;
+        sh:qualifiedValueShape [ sh:class brick:Hot_Water_Return_Temperature_Sensor ] ;
+        sh:qualifiedMinCount 1 ;
+        sh:qualifiedMaxCount 1 ;
+    ] ;
+    # applications
+    sh:property [
+        rdfs:label "Secondary HW supply or return flow" ;
+        sh:path brick:hasPoint ;
+        sh:qualifiedValueShape [
+            sh:or ( sh:class brick:Hot_Water_Supply_Flow_Sensor sh:class brick:Hot_Water_Return_Flow_Sensor )
+        ] ;
+        sh:qualifiedMinCount 0 ;
+        sh:qualifiedMaxCount 1 ;
+    ] ;
+    # TODO applications, for hybrid plants with condensing and non-condensing boilers.
+    sh:property [
+        rdfs:label "Intermediate HW supply temperature" ;
+        sh:path brick:hasPoint ;
+        sh:qualifiedValueShape [ sh:class brick:Hot_Water_Supply_Temperature_Sensor ] ;
+        sh:qualifiedMinCount 0 ;
+        sh:qualifiedMaxCount 1 ;
+    ] .
+
+:constant-primary-constant-secondary-loop a sh:NodeShape, owl:class, bmotif:HVAC ;
+    rdfs:label "Constant Primary Constant Secondary HW Loop" ;
+    sh:class brick:Hot_Water_Loop ;
+    sh:property [
+        rdfs:label "Constant Primary HW Loop" ;
+        sh:path brick:hasPart ;
+        sh:qualifiedValueShape [
+            sh:class brick:Hot_Water_Loop ;
+            sh:path brick:hasPart ;
+            sh:qualifiedValueShape [ sh:node components:constant-speed-pump ] ;
+            sh:qualifiedMinCount 1 ;
+        ] ;
+        sh:qualifiedMinCount 1 ;
+        sh:qualifiedMaxCount 1 ;
+    ] ;
+    sh:property [
+        rdfs:label "Constant Secondary HW Loop" ;
+        sh:path brick:hasPart ;
+        sh:qualifiedValueShape [
+            sh:class brick:Hot_Water_Loop ;
+            sh:path brick:hasPart ;
+            sh:qualifiedValueShape [ sh:node components:constant-speed-pump ] ;
+            sh:qualifiedMinCount 1 ;
+        ] ;
+        sh:qualifiedMinCount 1 ;
+        sh:qualifiedMaxCount 1 ;
+    ] .
```

Comment:
There's a small error in a few of these shapes -- `sh:path` needs to be in a PropertyShape, but the object of `sh:qualifiedValueShape` is a NodeShape. You'll need to nest a few of these properties inside a `sh:property` block:

```
    sh:property [
         rdfs:label "Constant Primary HW Loop" ;
         sh:path brick:hasPart ;
         sh:qualifiedValueShape [
             sh:class brick:Hot_Water_Loop ;
             sh:property [ # this line is new; puts the following 3 (existing) lines into a sh:property block
                sh:path brick:hasPart ;
                sh:qualifiedValueShape [ sh:node components:constant-speed-pump ] ;
                sh:qualifiedMinCount 1 ;
             ] # end of block
         ] ;
         sh:qualifiedMinCount 1 ;
         sh:qualifiedMaxCount 1 ;
     ] ;
```